src/app/components/ContactKeysTable.js

```diff
@@ -85,42 +128,64 @@ const ContactKeysTable = ({ model, setModel }) => {
-                                    }
-                                },
-                            model.isPGPInternal &&
-                                !isTrusted && {
-                                    text: c('Action').t`Trust`,
-                                    onClick() {
-                                        setModel({
-                                            ...model,
-                                            trustedFingerprints: [...model.trustedFingerprints, fingerprint]
-                                        });
-                                    }
-                                },
-                            model.isPGPInternal &&
-                                isTrusted && {
-                                    text: c('Action').t`Untrust`,
-                                    onClick() {
-                                        setModel({
-                                            ...model,
-                                            trustedFingerprints: model.trustedFingerprints.filter(
-                                                (f) => f !== fingerprint
-                                            )
-                                        });
-                                    }
-                                },
-                            model.isPGPExternal && {
+                            canBeActive && {
+                                text: c('Action').t`Use for sending`,
+                                onClick() {
+                                    const apiIndex = model.keys.api.findIndex(
+                                        (key) => key.getFingerprint() === fingerprint
+                                    );
+                                    const pinnedIndex = model.keys.pinned.findIndex(
+                                        (key) => key.getFingerprint() === fingerprint
+                                    );
+                                    const reOrderedApiKeys =
+                                        apiIndex !== -1 ? move(model.keys.api, apiIndex, 0) : model.keys.api;
+                                    const reOrderedPinnedKeys =
+                                        pinnedIndex !== -1
+                                            ? move(model.keys.pinned, pinnedIndex, 0)
+                                            : model.keys.pinned;
+                                    setModel({
+                                        ...model,
+                                        keys: { api: reOrderedApiKeys, pinned: reOrderedPinnedKeys }
+                                    });
+                                }
+                            },
+                            canBeTrusted && {
+                                text: c('Action').t`Trust`,
+                                onClick() {
+                                    const trustedFingerprints = new Set([...model.trustedFingerprints]);
```

Comment:
```suggestion
                                    const trustedFingerprints = new Set(model.trustedFingerprints);
```

Did you consider this?

Reply:
My bad, first time playing with sets